cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -19,9 +19,10 @@ const (
 )
 
 type Options struct {
-	Provider ProviderKind `yaml:"provider,omitempty"`
-	Path     string       `yaml:"path,omitempty"`
-	Module   string       `yaml:"module,omitempty"`
+	Provider         ProviderKind   `yaml:"provider,omitempty"`
+	Path             string         `yaml:"path,omitempty"`
+	Module           string         `yaml:"module,omitempty"`
+	DeploymentStacks map[string]any `yaml:"deploymentStacks,omitempty"`
```

Comment:
Given that we parse it into a concrete struct `azapi.deploymentStackOptions` at a later point in time of execution, would we prefer to simply do the parsing into a structured object here?

I would prefer to avoid `options map[string]any` in `bicep_provider.go` -- Bicep provider clearly knows what the options *should* look like, perhaps we would not want this type of error to be caught so late?

Reply:
I agree with you but we have a couple of constraints due to the package dependency graph that exist today that may cause circular dependencies.  Let me see if there are any other options available.

Reply:
This is good feedback but since this feature is only in `alpha` phase right now I'd prefer to push out as is till we collect more feedback from users.

Once we are ready to move into beta or GA we can update the config to reference a strongly typed configuration.

Reply:
Created #4399 to track this for upcoming sprint.